src/controls/BooleanControl.tsx

```diff
@@ -19,6 +19,8 @@ export function BooleanControl({
   config,
   description,
 }: ControlProps) {
+  if (!visible) return null
```

Comment:
I think in the case of a JSONForms Control we want the item fully removed from the DOM because if it's not visible, then the implication is its values shouldn't be evaluated or included in the form payload.